### PR TITLE
Istio Configuration in MII Model and DomainHomeOnPV model 

### PR DESCRIPTION
**Pull request by anpanigr**
Test case added to  verify ISTIO configuration  in 
  
  a) Model In Image  
  b) Domain Home On PV 

Additionally  (a) optimized the usage of utility methods  in the Test Classes instead of creating locally (b) modified istio installation scripts to to do some additional check to ensure proper installation.

Multi Node Cluster Jenkin
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-model-in-image-tests/1134

Kind Cluster 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/664

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+    if (KIND_REPO != null) {
+      String kindRepoImage = KIND_REPO + image.substring(TestConstants.OCR_REGISTRY.length() + 1);
+      logger.info("Using image {0}", kindRepoImage);
+      image = kindRepoImage;
+      isUseSecret = false;
+    }
+  }
+
+  /**
+   * Create a WebLogic domain using WLST in a persistent volume using wdt.
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:176]: It says both WLST and wdt, fix the description.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
+   * Create a configmap containing WLST script and property file.
+   * Create a Kubernetes job to create domain on persistent volume.
+   *
+   * @param wlstScriptFile       python script to create domain
+   * @param domainPropertiesFile properties file containing domain configuration
+   * @param pvName               name of the persistent volume to create domain in
+   * @param pvcName              name of the persistent volume claim
+   * @param namespace            name of the domain namespace in which the job is created
+   */
+  private void createDomainOnPVUsingWlst(Path wlstScriptFile, Path domainPropertiesFile,
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:373]: There is lot of duplicated code from ItDomainInPV which can be moved to common or util class.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    assertDoesNotThrow(() -> addLabelsToNamespace(opNamespace,labelMap));
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+  }
+
+  /**
+   * Create a domain using model-in-image model.
+   * Add istio configuration with default readinessPort 
+   * Do not add any AdminService under AdminServer configuration
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:126]: why is this?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    assertDoesNotThrow(() -> addLabelsToNamespace(domainNamespace,labelMap));
+    assertDoesNotThrow(() -> addLabelsToNamespace(opNamespace,labelMap));
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+  }
+
+  /**
+   * Create a domain using model-in-image model.
+   * Add istio configuration with default readinessPort
```
Comment (vanajamukkara) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:125]: add full stop at the end of each sentence.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java (hunk 3 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:126]: This is the requirement for istio. No Network Channel

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java (hunk 1 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:176]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java (hunk 4 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:125]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java (hunk 2 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:373]: @anpanigr how about this ^?

Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:373]: There are multiple such common functions which is repeated across many classes
 e.g.  createConfigMapForDomainCreation 
         createDomainOnPVUsingWlst
         createDomainJob
         createPV
         createPVC

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+  private final String adminServerName = "admin-server"; // do not modify
+  private String miiImage = null;
+  private static LoggingFacade logger = null;
+
+  private static Map<String, Object> secretNameMap;
+
+  /**
+   * Install Operator.
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   JUnit engine parameter resolution mechanism
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:94]: can you add a * at the beginning

Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:94]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+                .replicas(replicaCount)
+                .serverStartState("RUNNING"))
+            .configuration(new Configuration()
+                    .istio(new Istio()
+                         .enabled(Boolean.TRUE)
+                         .readinessPort(8888))
+                     .model(new Model()
+                         .domainType("WLS")
+                         .runtimeEncryptionSecret(encryptionSecretName))
+
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:293]: remove the blank line

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java [hunk 7]
```diff
@@ -107,7 +107,7 @@
   private static String dockerConfigJson = "";
 
   private static int replicaCount = 2;
-  private static final String domainUid = "miiaddcluster";
+  private static final String domainUid = "mii-add-cluster";
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java:110]: this is for line 125. I am not able to comment there. Can you add a *?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java (hunk 2 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:373]: yes, they all need to be moved to a util class, its lot of duplicate code. If you don't want to do as part of this PR, create a JIRA and it should be prioritized.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java (hunk 6 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioMiiDomain.java:293]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java (hunk 2 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:373]: Maggie has already has created method for createPV and createPVC and createConfigMapForDomainCreation in CommonTestUtils class.  I am trying to use the method from CommonTestUtils.  However   createDomainOnPVUsingWlst() is a more involved one, it calls createDomainJob() with another set of parameters, so we have to change the signature for createDomainOnPVUsingWlst() with more parameters.

Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItIstioDomainInPV.java:357]: Updated the ItIstioDomainInPV.java to use utility methods from utils/CommonTestUtils class
Add a JIRA to update other DomainInPV classes to do the same
https://jira.oraclecorp.com/jira/browse/OWLS-79135

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java (hunk 7 above)
Comment (anpanigr) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java:110]: done

**Pull request merged**